### PR TITLE
Support protocol 2.0 and php 8.3

### LINKED ISSUE #187: Add support for PHP 8.3
- Update CI to support PHP 8.3.
- Update README for the same.
- Internal discussion -> https://ably-real-time.slack.com/archives/CURL4U2FP/p1700733173822189

┆Issue is synchronized with this [Jira Task](https://ably.atlassian.net/browse/ECO-4572) by [Unito](https://www.unito.io)

### LINKED ISSUE #190: [REST] update to protocol 2.0
- Update agent http header `X-Ably-Version` to 2.0

┆Issue is synchronized with this [Jira Task](https://ably.atlassian.net/browse/ECO-4699) by [Unito](https://www.unito.io)

### LINKED ISSUE #192: Fix idempotent rest publishing
- Fix incomplete https://github.com/ably/ably-php/issues/56
- Need to check missing spec as a part of https://github.com/ably/ably-php/pull/58

┆Issue is synchronized with this [Jira Task](https://ably.atlassian.net/browse/ECO-4704) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by sacOO7**
Fixed #190 
Fixed #187 
Fixed #192

### CODE REVIEW

File: src/Defaults.php [hunk 1]
```diff
@@ -2,7 +2,7 @@
 namespace Ably;
 
 class Defaults {
-    const API_VERSION = '1.1';
+    const API_VERSION = '2.0';
```
Comment (ttypic) [inline src/Defaults.php:5]: Is it protocol version? Should it be `2` then?

Comment (sacOO7) [reply src/Defaults.php:5]: Let me check !

Comment (sacOO7) [reply src/Defaults.php:5]: Yeah, it should be `2`. Seems I followed https://github.com/ably/specification/pull/88/files.

Comment (sacOO7) [reply src/Defaults.php]: Silly mistake !

Comment (sacOO7) [reply src/Defaults.php]: updated

**Comment by sacOO7** [comment]
> LGTM

Thanks for the approval 👍

**Pull request merged**